src/vsg/vk/Context.cpp

```diff
@@ -244,7 +245,7 @@ void Context::reserve(const ResourceRequirements& requirements)
     if (required_maxSets > 0 || !required_descriptorPoolSizes.empty())
     {
         getDescriptorPoolSizesToUse(required_maxSets, required_descriptorPoolSizes);
-        if (required_maxSets > 0 && !required_descriptorPoolSizes.empty())
+        if (required_maxSets > 0 || !required_descriptorPoolSizes.empty())
```

Comment:
The recent changes were made because we were seeing Vulkan debug errors when maxSets was 0 or the pool sizes vector was empty so checking both using an && was required.